bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/internal/WebClientFactoryImpl.java

```diff
@@ -362,4 +366,42 @@ private SslContextFactory createSslContextFactory() {
 
         return sslContextFactory;
     }
+
+    @Override
+    public HTTP2Client createHttp2Client(String consumerName) {
+        return createHttp2Client(consumerName, null);
+    }
+
+    @Override
+    public HTTP2Client createHttp2Client(String consumerName, @Nullable SslContextFactory sslContextFactory) {
+        logger.debug("http client for consumer {} requested", consumerName);
+        checkConsumerName(consumerName);
+        return createHttp2ClientInternal(consumerName, sslContextFactory);
+    }
+
+    private HTTP2Client createHttp2ClientInternal(String consumerName, @Nullable SslContextFactory sslContextFactory) {
+        try {
+            logger.debug("creating HTTP/2 client for consumer {}", consumerName);
+
+            try {
+                PreEncodedHttpField field = new PreEncodedHttpField(HttpHeader.C_METHOD, "PUT");
+                ByteBuffer bytes = ByteBuffer.allocate(32);
+                field.putTo(bytes, HttpVersion.HTTP_2);
+            } catch (Exception e) {
+                throw new HttpClientInitializationException("Jetty HTTP/2 hpack module not loaded", e);
+            }
```

Comment:
Does this really need to be checked every time? I think when the hpack bundle is stopped after this check would it still throw some exception when creating or using the client?

Reply:
The issue is that the lack of HTTP/2 hpack encoding is not detected when CREATING the client. Indeed it is not normally detected when doing HTTP/ GET calls with commonly used headers (because these common cases are pre-encoded without needin hpack). The problem only actually arises when the client makes more 'exotic' calls such as PUT, or uses less common headers, which require encoding on the fly; and at this point the lack cayses an exception.

So my intention is to force throw the exception on client creation. In theory the check could be done one time only on creating the first client. The only problem with that is that you could see the exception originating from only the first binding that creates a client (whichever it might be) and not from any subsequent binding creating a client; so you might conclude that the issue was randomly related to that one first binding, rather than being a common issue in the core.

Reply:
PS I just pushed a minor optimisation..